src/api/time/IndependentTimeContext.js

```diff
@@ -20,18 +20,44 @@
  * at runtime from the About dialog for additional information.
  *****************************************************************************/
 
-import TimeContext from "./TimeContext";
+import TimeContext, { TIME_CONTEXT_EVENTS, TIME_CONTEXT_METHODS } from "./TimeContext";
 
 /**
  * The IndependentTimeContext handles getting and setting time of the openmct application in general.
  * Views will use the GlobalTimeContext unless they specify an alternate/independent time context here.
  */
 class IndependentTimeContext extends TimeContext {
-    constructor(globalTimeContext, key) {
+    constructor(globalTimeContext, objectPath) {
         super();
-        this.key = key;
-
+        this.unlisteners = [];
         this.globalTimeContext = globalTimeContext;
+        this.objectPath = objectPath;
+        this.refreshContext = this.refreshContext.bind(this);
+        this.resetContext = this.resetContext.bind(this);
+
+        TIME_CONTEXT_METHODS.forEach((functionName) => {
```

Comment:
This is a neat solution. I'm putting myself in the position of someone trying to debug this though, and it adds a later of indirection that might make life difficult. I wonder if we should just override each of these methods on the `IndependentTimeConductor` and add this logic to each method explicitly, so it's clear what's happening.

eg.
``` JavaScript
bounds() {
    if (this.upstreamTimeContext !== undefined) {
        this.upstreamTimeContext.bounds(...arguments);
    } else {
        super.bounds(...arguments)
    }
}
```

Reply:
Right. That 's what I did initially, but the code was identical, so I did this. I do agree that readability is a problem with the current solution. I'll change it.

Reply:
Done